src/components/CompareResults/beta/ResultsView.tsx

```diff
@@ -64,44 +63,44 @@ function ResultsView(props: ResultsViewProps) {
 
   const sectionStyles = SearchContainerStyles(themeMode, compareView);
 
-  const location = useLocation();
   const { dispatchFetchCompareResults } = useFetchCompareResults();
   const { searchByRevisionOrEmail } = useHandleChangeSearch();
 
   useEffect(() => {
     document.title = title;
   }, [title]);
 
-  useEffect(() => {
-    const urlSearchParams = new URLSearchParams(location.search);
-    const repos = urlSearchParams.get('repos')?.split(',');
-    const revs = urlSearchParams.get('revs')?.split(',');
-    const framework = urlSearchParams.get('framework');
+  const repos = searchParams.get('repos');
+  const revs = searchParams.get('revs');
+  const framework = searchParams.get('framework');
 
+  useEffect(() => {
     if (revs && repos) {
+      const revsArray = revs.split(',');
+      const reposArray = repos.split(',');
       void dispatchFetchCompareResults(
-        repos as Repository['name'][],
-        revs,
+        reposArray as Repository['name'][],
+        revsArray,
         framework as string,
       );
 
       /*
       On component mount, use the repos and revs in hash to search for the base and new revisions. Store the results in state via the SelectedRevisionsSlice: see extra reducer, fetchRevisionsByID. Now can always display the selected revisions despite page refresh or copying and pasting url
       */
-      revs.forEach((rev, index) => {
+      revsArray.forEach((rev, index) => {
         void searchByRevisionOrEmail(
-          repos[index] as Repository['name'],
+          reposArray[index] as Repository['name'],
           rev,
           'base',
         );
         void searchByRevisionOrEmail(
-          repos[index] as Repository['name'],
+          reposArray[index] as Repository['name'],
           rev,
           'new',
         );
       });
     }
-  }, []);
+  }, [repos, revs, framework]);
```

Comment:
I believe this will make the back button work properly, because now the search will be repeated as soon as the dependencies change.

Maybe the second part should be separated though, because it doesn't use `framework`... We can do that after sumair patch that touches also this area.